src/adapter/src/catalog.rs

```diff
@@ -2830,6 +2901,92 @@ impl CatalogItem {
             | CatalogItem::Connection(_) => false,
         }
     }
+
+    pub(crate) fn to_serialized(&self) -> SerializedCatalogItem {
+        match self {
+            CatalogItem::Table(table) => SerializedCatalogItem::V1 {
+                create_sql: table.create_sql.clone(),
+            },
+            CatalogItem::Log(_) => unreachable!("builtin logs cannot be serialized"),
+            CatalogItem::Source(source) => {
+                assert!(
+                    match source.data_source {
+                        DataSourceDesc::Introspection(_) => false,
+                        _ => true,
+                    },
+                    "cannot serialize introspection/builtin sources",
+                );
+                SerializedCatalogItem::V1 {
+                    create_sql: source.create_sql.clone(),
+                }
+            }
+            CatalogItem::View(view) => SerializedCatalogItem::V1 {
+                create_sql: view.create_sql.clone(),
+            },
+            CatalogItem::MaterializedView(mview) => SerializedCatalogItem::V1 {
+                create_sql: mview.create_sql.clone(),
+            },
+            CatalogItem::Index(index) => SerializedCatalogItem::V1 {
+                create_sql: index.create_sql.clone(),
+            },
+            CatalogItem::Sink(sink) => SerializedCatalogItem::V1 {
+                create_sql: sink.create_sql.clone(),
+            },
+            CatalogItem::Type(typ) => SerializedCatalogItem::V1 {
+                create_sql: typ.create_sql.clone(),
+            },
+            CatalogItem::Secret(secret) => SerializedCatalogItem::V1 {
+                create_sql: secret.create_sql.clone(),
+            },
+            CatalogItem::Connection(connection) => SerializedCatalogItem::V1 {
+                create_sql: connection.create_sql.clone(),
+            },
+            CatalogItem::Func(_) => unreachable!("cannot serialize functions yet"),
+        }
```

Comment:
Couldn't we replace the implementation with `self.clone().into_serialized()`? We need to clone the `create_sql` anyways, and whether we do it manually or through the outer type seems equivalent, without checking the details here.

Reply:
I wanted to avoid cloning all of `self` since some of the fields can get rather large and we would throw them all away.